src/systems/system.C

```diff
@@ -288,7 +295,25 @@ void System::init_data ()
    this->init_matrices();
 }
 
+void System::reinit_mesh ()
```

Comment:
So at this level we didn't need anything different than System::init() except for the removed assert()?

So:

A. Have we thought enough about how this should interact with derived systems?  I thought we'd discussed possibly wanting reinit_mesh() to be virtual, so derived systems would have a chance to react to a change in mesh?

B. Should System::init() just be that assert followed by a reinit_mesh()?  Based on your design right now the answer is "of course it should", but the answer to (A) could make this more questionable.

Reply:
> So at this level we didn't need anything different than System::init() except for the removed assert()?

Yes, this seems to be the case. Calling solve() after moving behaves as expected.

> 
> So:
> 
> A. Have we thought enough about how this should interact with derived systems? I thought we'd discussed possibly wanting reinit_mesh() to be virtual, so derived systems would have a chance to react to a change in mesh?

Wouldnt this overlap with the user defined initialization function we call at the end of the reinit_mesh() function ?

> 
> B. Should System::init() just be that assert followed by a reinit_mesh()? Based on your design right now the answer is "of course it should", but the answer to (A) could make this more questionable.

The question is whether there is anything inside System::init_data that does not need execution when we only mean to reinit the mesh. I think there isnt, because almost every data structure owned by System relies on its mesh to provide the scaffolding.

Reply:
After some thought - I don't think making reinit_mesh() virtual is an overlap here.  In some sense the "user" of a user_initialization() isn't the class designer, it's the class instantiator.  So we'd want the subclass designer to be able to overload reinit_mesh() without messing with any separate user initialization that's been attached.

Plus, what the hell, this is one function call per replacing-the-whole-freaking-mesh, so if it takes an extra nanosecond to go to the vtable that's worth some added flexibility.

Reply:
Sounds reasonable.